PROBot/UserSettings.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System.IO;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace PROBot
+{
+    public static class UserSettings
+    {
+        private static SettingsCache _settings;
+
+        public static bool AutoReconnect
+        {
+            get { return _settings.AutoReconnect; }
+            set
+            {
+                if (_settings.AutoReconnect != value)
+                {
+                    _settings.AutoReconnect = value;
+                    _settings.Save();
+                }
+            }
+        }
+
+        public static bool AvoidStaff
+        {
+            get { return _settings.AvoidStaff; }
+            set
+            {
+                if (_settings.AvoidStaff != value)
+                {
+                    _settings.AvoidStaff = value;
+                    _settings.Save();
+                }
+            }
+        }
+
+        public static bool AutoEvolve
+        {
+            get { return _settings.AutoEvolve; }
+            set
+            {
+                if (_settings.AutoEvolve != value)
+                {
+                    _settings.AutoEvolve = value;
+                    _settings.Save();
+                }
+            }
+        }
+
+        public static string LastScript
+        {
+            get { return _settings.LastScript; }
+            set
+            {
+                if (_settings.LastScript != value)
+                {
+                    _settings.LastScript = value;
+                    _settings.Save();
+                }
+            }
+        }
+
+        static UserSettings()
+        {
+            try
+            {
+                if (File.Exists("Settings.json"))
+                {
+                    string fileText = File.ReadAllText("Settings.json");
+                    JObject json = JsonConvert.DeserializeObject(fileText) as JObject;
+                    _settings = JsonConvert.DeserializeObject<SettingsCache>(json.ToString());
+                    return;
```

Comment:
You can remove the `return` here and move `_settings = new SettingsCache();` inside the `catch {}`

Reply:
It needs to create a new setting if the file does not exist, that's why I did it that way.

Reply:
If you move the new settings creation inside the catch, the behaviour stays the same:
```cs
// ...
catch
{
    _settings = new SettingsCache();
}
```

Reply:
If you delete the Settings file and try it again, you'll see that _settings will be null if you do that.

Reply:
Oh, indeed, I forgot about the if. We can keep it like this then.